### PR TITLE
Update roadmap for 2021

### PR DESCRIPTION
**Pull request by bboreham**
Implemented since last update:
 - Per-tenant retention,
 - Soft Multitenancy,
 - Prometheus metadata support,
 - Bulk loading Thanos data,
 - Alertmanager sharding

I also made some changes for better readability.

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -27,22 +29,22 @@ tenants:
...
 
-Currently, we only support a single retention period for all metrics and tenants. For most operators, the ability to set per tenant retention and also custom retention for subsets of metrics is important. We will add support per tenant and metric retention policies. Also, we currently store all the samples we ingested, and there is no way to reduce the resolution for the metrics. We plan to add downsampling to allow users to store less data when needed.
+## Per-metric retention
 
-## Soft Multitenancy
+Cortex blocks storage supports deleting all data for a tenant after a time period (e.g. 3 months, 1 year), but we would also like to have custom retention for subsets of metrics (e.g. delete server metrics but retain business metrics).
 
-Currently our multitenancy allows a tenant to view _all_ their metrics and only their metrics. There is no way for an "admin" tenant to view all the metrics in the system but for particular teams to only view theirs. This is another feature we plan to add into Cortex.
-
-## Exemplar and Prometheus metadata support
```
Comment (pracucci) [inline docs/roadmap.md:38]: > also add persistence of the metadata (right now it's only in-mem)

This has not been done, but not sure if we want it. At least, doesn't look a priority until Prometheus will keep metadata only in-memory as well.

**Review by pracucci — APPROVED** [review]
Thanks Bryan!

File: docs/roadmap.md (hunk 1 above)
Comment (bboreham) [reply docs/roadmap.md:38]: Good point.  For consistency, metadata should match the point in history where you are looking, but it doesn't strike me as a hi-priority feature.

File: docs/roadmap.md [hunk 2]
```diff
@@ -27,22 +29,22 @@ tenants:
...
+## Exemplar support
+[Exemplars](https://docs.google.com/document/d/1ymZlc9yuTj8GvZyKz1r3KDRrhaOjZ1W1qZVW_5Gj7gA/edit)
+let you link metric samples to other data, such as distributed tracing.
+As of early 2021 Prometheus will collect exemplars and send them via remote write, but Cortex needs to be extended to handle them.
 
 ## Bulk loading historical data
-
-This is another highly requested features. There is currently no way to backfill the existing data in local Prometheus to Cortex. The plan is to add an API for users to ship the TSDB blocks to Cortex and a side-car / command to do this.
+This is another highly requested feature.  As of 1.8.0 there is a tool to import Thanos blocks into Cortex, but not for basic Prometheus.
```
Comment (pstibrany) [inline docs/roadmap.md:46]: It works for basic Prometheus blocks too... One needs to upload blocks to bucket manually (under tenant prefix), and then run `thanosconvert` tool to fix metadata on such blocks.

Comment (bboreham) [reply docs/roadmap.md:46]: Thanks. I guess this can come off the roadmap then.

**Comment by bboreham** [comment]
Updated in line with review comments.

**Pull request merged**